content/browser/secure_embed_connector_impl.cc

```diff
@@ -89,6 +90,51 @@ void SecureEmbedConnectorImpl::FocusInEmbedder(FocusOperation focus_op) {
   }
 }
 
+FrameTree* SecureEmbedConnectorImpl::GetFocusedFrameTree() {
+  if (!embedder_web_contents_) {
+    return &guest_web_contents_->GetPrimaryFrameTree();
+  }
+
+  return static_cast<WebContentsImpl*>(embedder_web_contents_.get())
+      ->GetFocusedFrameTree();
+}
+
+void SecureEmbedConnectorImpl::SetFocusedFrameTree(
+    FrameTree* frame_tree_to_focus) {
+  if (!embedder_web_contents_) {
+    return;
+  }
+
+  auto* embedder_web_contents =
+      static_cast<WebContentsImpl*>(embedder_web_contents_.get());
+
+  // Update focused frame tree stored in the embedder.
+  embedder_web_contents->SetFocusedFrameTree(frame_tree_to_focus);
+
+  // Ensure that outer frame trees are focused.
+  embedder_web_contents->GetPrimaryFrameTree().FocusOuterFrameTrees();
+
+  // Ensure that the embedder's page has focus so that it can display active UI
+  // and therefore the embedded plugin is also active.
+  embedder_web_contents->GetPrimaryMainFrame()
+      ->GetRenderWidgetHost()
+      ->SetPageFocus(true);
+}
+
+void SecureEmbedConnectorImpl::ClearFocusedFrameTreeIfNecessary() {
```

Comment:
Can we name this `ClearFocusOnInnerWebContents`? I think the current name is a bit confusing - it does not tell if the focus on an outer frame should be cleared.